apps/studio/components/interfaces/Auth/Policies/AIPolicyEditorPanel/RLSCodeEditor.tsx

```diff
@@ -0,0 +1,120 @@
+import Editor, { OnChange, OnMount } from '@monaco-editor/react'
+import { editor } from 'monaco-editor'
+import { MutableRefObject, useRef } from 'react'
+import { cn } from 'ui'
+
+import { alignEditor } from 'components/ui/CodeEditor'
+import { Markdown } from 'components/interfaces/Markdown'
+
+// [Joshen] Is there a way we can just have one single MonacoEditor component that's shared across the dashboard?
+// Feels like we're creating multiple copies of Editor
+
+interface RLSCodeEditorProps {
+  id: string
+  defaultValue?: string
+  onInputChange?: (value?: string) => void
+  wrapperClassName?: string
+  className?: string
+  value?: string
+  editorRef: MutableRefObject<editor.IStandaloneCodeEditor | null>
+}
+
+const placeholderText = `
+CREATE POLICY _name_ ON _table_name_\n
+[ AS { PERMISSIVE | RESTRICTIVE } ]\n
+[ FOR { ALL | SELECT | INSERT | UPDATE | DELETE } ]\n
+[ TO _role_name_ ]\n
+[ USING ( _using_expression_ ) ]\n
+[ WITH CHECK ( _check_expression_ ) ];
+`.trim()
+
+const RLSCodeEditor = ({
+  id,
+  defaultValue,
+  wrapperClassName,
+  className,
+  value,
+  editorRef,
+}: RLSCodeEditorProps) => {
+  const hasValue = useRef<any>()
+
+  const onMount: OnMount = async (editor, monaco) => {
+    editorRef.current = editor
+    alignEditor(editor)
+
+    hasValue.current = editor.createContextKey('hasValue', false)
+
+    const placeholder = document.querySelector('.monaco-placeholder') as HTMLElement | null
+    if (placeholder) placeholder.style.display = 'block'
+
+    editor.addCommand(
+      monaco.KeyCode.Tab,
+      () => {
+        editor.executeEdits('source', [
+          {
+            // @ts-ignore
+            identifier: 'add-placeholder',
+            range: new monaco.Range(1, 1, 1, 1),
+            text: placeholderText.split('\n\n').join('\n'),
```

Comment:
Instead of using the placeholder text directly, it might be nice to make it a bit more usable – currently the user has to delete a lot of `[` / `]`. Something like:

```
CREATE POLICY _name_ ON _table_name_
AS PERMISSIVE
FOR ALL
TO _role_name_
USING ( _using_expression_ )
WITH CHECK ( _check_expression_ );
```

(this is just opinion though)

Reply:
hmm good point though - that plus while this template is from the postgres docs, postgres newbies wouldn't know what the square brackets represent (which is them being optional)

Reply:
wondering if it might make sense adding comments to annotate the template